### DISCUSSION TITLE
Disable new translations?

### DISCUSSION THREAD

**Discussion by sitenzo** [andrewdwallo/transmatic#18] [Q&A] [answered]
Hi,

Is it possible to disable the translator when the .env configuration is not set?
The Scenario : 

- Development and making translations are on my local environment.
- When deploying i remove the AWS keys.
- When translation is not set it will then throw a *CredentialsException*
  - instead of the Exception i would like to fallback to text that would need a translate and a info or error log in my log file of what needs to be translated
  - Then the user does not see a error 500 and i know that i missed a translation somewhere.
  
Is this something that is already possible?
  
Thanks for this wonderful library.

**Comment by andrewdwallo** [comment, marked as answer]
Hi @sitenzo,

Thank you for reaching out and for your kind words about the package!

Based on the scenario you described, where you want to disable the translator in production by removing the AWS keys, there's a straightforward solution that aligns with your requirements.

You can modify the AWS configuration used by the AWS Service Provider for Laravel as documented in their [Configuration section](https://github.com/aws/aws-sdk-php-laravel?tab=readme-ov-file#configuration). By publishing the `aws.php` config file to your Laravel application, you gain the ability to customize the AWS SDK's behavior based on your environment.

For your specific case, you can update the credentials configuration in the `config/aws.php` file from this:
```php
'credentials' => [
    'key' => env('AWS_ACCESS_KEY_ID', ''),
    'secret' => env('AWS_SECRET_ACCESS_KEY', ''),
],
```

To simply:
```php
'credentials' => false,
```

This change will instruct the AWS SDK not to attempt any translation when the credentials are explicitly set to false. Consequently, the translator will not throw any `CredentialsException` or log any errors related to AWS translation attempts. Transmatic is designed to return the original, untranslated text by default when translation is not performed, which fits your requirement perfectly.

Furthermore, to implement the logging functionality you mentioned — logging untranslated texts for later review — you can create a custom `Translator` class as described [here](https://github.com/andrewdwallo/transmatic#custom-translation-service-integration) to check if AWS credentials are set to false (or are nonexistent) in your `aws.php` configuration or elsewhere. Although I'm not entirely certain, I believe the AWS SDK API may offer methods to check the presence of these credentials. Thus, it's feasible to determine the absence of credentials using either the API directly or the configuration file. Within this class, you can implement the desired logging functionality for texts that require translation.

**Reply by sitenzo** [reply]
Thanks for the detailed response.
